draft-ietf-lake-edhoc.md

```diff
@@ -368,7 +369,29 @@ If connection identifiers are used by an application protocol for which EDHOC es
 
 ### Use of Connection Identifiers with OSCORE {#ci-oscore}
 
-For OSCORE, the choice of connection identifier results in the endpoint selecting its Recipient ID, see Section 3.1 of {{RFC8613}}, for which certain uniqueness requirements apply, see Section 3.3 of {{RFC8613}}. Therefore, the Initiator and the Responder MUST NOT select connection identifiers such that it results in same OSCORE Recipient ID. Since the Recipient ID is a byte string and an EDHOC connection identifier is either a CBOR byte string or a CBOR integer, care must be taken when selecting the connection identifiers and converting them to Recipient IDs. A mapping from EDHOC connection identifier to OSCORE Recipient ID is specified in {{edhoc-to-oscore}}.
+For OSCORE, the choice of connection identifier results in the endpoint selecting its Recipient ID, see Section 3.1 of {{RFC8613}}, for which certain uniqueness requirements apply, see Section 3.3 of {{RFC8613}}. Therefore, the Initiator and the Responder MUST NOT select connection identifiers such that it results in same OSCORE Recipient ID. Since the connection identifier is a byte string, it is converted to an OSCORE Recipient ID equal to the byte string.
+
+For example, a byte string valued C_I equal to 0xFF (0x41FF in CBOR encoding) is converted to a (typically client) Responder ID equal to 0xFF.
+
+### Representation of Byte String Identifiers {#bstr-repr}
+
+The integers -24, ..., 23 are all CBOR encoded as one byte, see {{fig-int-one-byte}}.
+
+~~~~~~~~~~~
+Integer:               -24   -23  ...   -2    -1     0     1  ...   23
+CBOR encoding (1 byte): 37    36  ...   21    20    00    01  ...   17
+~~~~~~~~~~~
+{: #fig-int-one-byte title="One-Byte CBOR Encoded Integers"}
+{: artwork-align="center"}
+
+To allow more byte string identifiers that are only one byte on the wire, certain byte strings are defined to  have integer representations in the interval -24, ... ,23. These byte strings are those for which there is an integer in the interval -24, ..., 23 that happens to have this byte string as its CBOR encoding, see {{fig-bstr-as-int}} and compare {{fig-int-one-byte}}.
+
```

Comment:
Is there a reason to limit this representation only to the interval -24..23?

If it rather considered all the integer values:

* There will be several more identifiers that can be transported as integers, again with 1 byte less overhead compared to the byte string representation. For example, the identifier 0x1839 can be transported as 0x1839 (CBOR integer 57) rather than 0x421839 (CBOR byte string).

* I believe it would be overall simpler for an implementation, that would simply have to check about integer vs. byte string, rather than specifically integer -24..23 vs. byte string.

The same applies for 'kid' values when 'kid' is used in ID_CRED_X.

Reply:
Without commenting on this proposal, I don't think we should optimise the sheer number of short identifiers.
There needs to be a balance here between what is easily specified, understood and coded. 

IMO, as long as we can have a number of 1-byte connection identifiers and key identifiers in EDHOC (to comply with the LAKE requirements) and support for the empty byte string in OSCORE (to reduce bytes in the most frequent message exchange) the main effort should be spent on making the EDHOC specification and the conversion between EDHOC and OSCORE identifiers as simple as possible. 

@marco-tiloca-sics Have you written down the procedure for reversing the representation, and for mapping back and forth between EDHOC and OSCORE?

One concern I have (with #267 in general) is with the pre-pending of connection identifiers done at transport layer. If we introduce special EDHOC representations, does the transport layer also need to know about that? (This was one reason for doing #274.)

Reply:
The proposal to apply the mapping for all integers is now included. I agree that the user of transport needs to provide the transport-ready representation, including prepended identifier, this is also noted in the update.

Reply:
Now reverted to one-byte integers only.